### PR TITLE
Fix: replace homemade release script with goreleaser

### LINKED ISSUE #5347: checksums are wrong in v1.7.0
https://github.com/kubevela/kubevela/releases/tag/v1.7.0

```console
$ sha256sum vela-v1.7.0-linux-amd64.tar.gz 
e67e499c124dde558de90f983dd05444b53981c5c3e53eaaff9a98b552de6816  vela-v1.7.0-linux-amd64.tar.gz

$ grep vela-v1.7.0-linux-amd64.tar.gz sha256sums.txt 
f8816b118ce9d2f508a74af91e556884568f7421c3e4b34a02abdd2ab767247e  vela-v1.7.0-linux-amd64.tar.gz
6311ebbe2139e9e5219c9d39eec07d20b7fb9cb1e98b4ce3eb2233b118b3a642  kubectl-vela-v1.7.0-linux-amd64.tar.gz

$ sha256sum --ignore-missing -c sha256sums.txt
vela-v1.7.0-linux-amd64.tar.gz: FAILED
sha256sum: WARNING: 1 computed checksum did NOT match
sha256sum: sha256sums.txt: no file was verified
```

### PR DESCRIPTION
**Pull request by chivalryq**
Signed-off-by: Qiaozp <qiaozhongpei.qzp@alibaba-inc.com>

### Replacement

This PR introduce goreleaser too KubeVela's CLI and kubectl plugin release process. Goreleaser will take care of these things which is done by script in GitHub Action.

1. Build artifacts
2. Archive the binary and other file needed.
3. Upload all archives and `sha256sums.txt` to GitHub release page.

### Enhancement

Add windows/arm64 artifacts

Fixes #5347

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

The whole process may influence several things:

- [x] Artifact function: tested manually.
- [x] GitHub Action "release": 
  - [x] Upload artifacts to OSS: test manually.
  - [x] Upload to GitHub release page: tested in this [release](https://github.com/chivalryq/kubevela/releases/tag/v1.7.3) in my fork.
  - [ ] Other parts in release (Update krew-index and update Homebrew formula): not tested but the influence is limited.
- [x] Installation: test by install a mocked version.
```shell
curl -fsSl https://static.kubevela.net/script/install.sh | bash -s 1.7.0-d1
```

### CODE REVIEW

File: .goreleaser.yaml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    goos:
+      - linux
+      - windows
+      - darwin
+    goarch:
+      - amd64
+      - arm64
+    main: ./references/cmd/cli/main.go
+    ldflags:
+      - -s -w -X github.com/oam-dev/kubevela/version.VelaVersion={{ .Version }} -X github.com/oam-dev/kubevela/version.GitRevision={{.Commit}}
```
Comment (wonderflow) [inline .goreleaser.yaml:15]: `git-{{.Commit}}` ?

File: .goreleaser.yaml [hunk 2]
```diff
@@ -0,0 +1,84 @@
+# This is an example .goreleaser.yml file with some sensible defaults.
+# Make sure to check the documentation at https://goreleaser.com
+builds:
+  - id: vela-cli
+    binary: vela
+    goos:
+      - linux
+      - windows
```
Comment (wonderflow) [inline .goreleaser.yaml:8]: will it generate arm64 for windows?

Comment (chivalryq) [reply .goreleaser.yaml:8]: Yes

File: .goreleaser.yaml (hunk 1 above)
Comment (chivalryq) [reply .goreleaser.yaml:15]: Fixed

**Review by Somefive — APPROVED** [review]

**Pull request merged**